sublime_linter.py

```diff
@@ -146,6 +146,16 @@ def on_pre_close(self, view):
 
         queue.cleanup(vid)
 
+        # Remove bid from persist.errors if it's the last view on the buffer
+        bid = view.buffer_id()
+        buffers = []
+        for w in sublime.windows():
+            for v in w.views():
+                buffers.append(v.buffer_id())
+
+        if buffers.count(bid) is 1:
```

Comment:
just make it `<= 1` bc shit happens

Reply:
Well no, it can also be 0 for some reason (shit happens) and we don't want that because `del persist.errors[bid]` will fail.

Reply:
And what shit happened?

Reply:
Since this is the only place where we call del[bid] it means we call it twice, the first time when buffers.count is 1 and then a second time when buffers.count is 0

Reply:
Good question. Will have a look.

Reply:
Okay it probably comes from views we never linted before. I thought we have a precheck here. But now, if you look above, we usually use `pop(vid, None)` so that we never throw (bc shit happens)

Reply:
So if you're more into defense, you probably do the `<= 1` check with a `pop(bid, None)`. And otherwise if you're really sure, you just commit what you got.

Reply:
Probably a good place to be defensive. Cleaning up is a "best effort" thing anyway considering we run in a kinda messy environment.